app/workers/zync_worker.rb

```diff
@@ -220,8 +222,7 @@ def perform(event_id, notification)
       http_put(notification_url, notification, event_id)
     end
   rescue UnprocessableEntityError
-    publish_dependencies_events(event_id)
-    raise
+    raise if last_attempt? || sync_dependencies(event_id).blank?
```

Comment:
Isn't it`unless`?

Reply:
No. if it's the last retry attempt or there's no dependencies to sync before, then we cannot really rescue from `UnprocessableEntityError`; better raise the error.